### PR TITLE
Fly3D

### PR DESCRIPTION
**Pull request by Z3R0GT**
"Node config for CharacterBody3D"

in this node, the "Player" can fly :D

Issue #65

### CODE REVIEW

**Comment by Z3R0GT** [comment]
wait for review :D 👍

File: addons/nodot/characters/Fly3D.gd [hunk 1]
```diff
@@ -0,0 +1,143 @@
+## This node is a mechanic for fly player (experimental)
+class_name Fly3D extends Node3D
+
+## This signal call in "_physics_process()" of player: $(Fly3D).emit_signal("config_fly", self, $(model of player))
+signal config_fly(Parent, model)
```
Comment (krazyjakee) [inline addons/nodot/characters/Fly3D.gd:5]: Would prefer to use the get_parent setup like the other movement nodes. Sending a signal means we would have to modify the character controller and break modularity.

Also lower case the parent argument

File: addons/nodot/characters/Fly3D.gd [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+@export_category("Configs Funcs")
```
Comment (krazyjakee) [inline addons/nodot/characters/Fly3D.gd:7]: Please use full words "functions" though these also aren't functions.

File: addons/nodot/characters/Fly3D.gd [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+## Can rotate model of player
+@export var can_rotate = true
+## when scene start, move is automatic
+@export var automatic_move = false
+## Check when it can start to fly
+@export var limit_high = 2.0
```
Comment (krazyjakee) [inline addons/nodot/characters/Fly3D.gd:13]: What is the use case for this?

File: addons/nodot/characters/Fly3D.gd [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+## Temp space from default speed
+var temp_default_speed
+
+func _enter_tree():
+	temp_default_speed = default_speed
+	
+	if get_parent() is CharacterBody3D:
+		connect("config_fly", _fly)
+	else:
+		print_debug("Parent must have CharacterBody3D, no: ", get_parent())
```
Comment (krazyjakee) [inline addons/nodot/characters/Fly3D.gd:43]: Please see the other nodes for how we are handling node config warnings.

File: addons/nodot/characters/Fly3D.gd [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+		## send local velocity
+		Parent.set_velocity(velocity)
+		## local move and slide
+		Parent.move_and_slide()
+	else:
+		return
+	
+func get_input(model):
+	## Config new controlls
+	var input_dir = Input.get_vector("ui_fly_left", "ui_fly_right", "ui_fly_foward", "ui_fly_back")
```
Comment (krazyjakee) [inline addons/nodot/characters/Fly3D.gd:61]: Check the other nodes for action naming conventions and setup. These actions need to be automatically defined with default controls for both keyboard and controller.

**Review by krazyjakee — CHANGES_REQUESTED** [review]
This wasn't testable so I haven't manually tested it but the code needs some work. Definitely take time to look over the other movement nodes to see what ia missing.

**Pull request closed without merging**